src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -302,6 +315,35 @@ private string GetAnalyzerPath(AnalyzerReference analyzerReference)
             return analyzerReference.FullPath;
         }
 
+        protected override void ApplyCompilationOptionsChanged(ProjectId projectId, CompilationOptions options)
+        {
+            if (projectId == null)
+            {
+                throw new ArgumentNullException(nameof(projectId));
+            }
+
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            GetProjectData(projectId, out var hostProject, out var hierarchy, out var project);
+            foreach (Configuration configuration in project.ConfigurationManager)
+            {
+                switch (hostProject.Language)
+                {
+                    case LanguageNames.CSharp:
+                        var csharpProperties = (VSLangProj80.CSharpProjectConfigurationProperties3)configuration.Object;
+
+                        csharpProperties.AllowUnsafeBlocks = ((CSharpCompilationOptions)options).AllowUnsafe;
```

Comment:
I'd be curious what this looks like if we use IVsBuildPropertyStorage instead. It means we don't have to be looping through configurations/platforms, and would also have VB support.

@Pilchie/@davkean: any concerns with trying IVsBuildPropertyStorage?